File: solutions/python/allergies/1/allergies.py
ALLERGEN = {"cats": 128,
            "pollen": 64,
            "chocolate": 32,
            "tomatoes": 16,
            "strawberries": 8,
            "shellfish": 4,
            "peanuts": 2,
            "eggs": 1}


class Allergies:

    def __init__(self, score):
        self._allergens = []
        for key, value in ALLERGEN.items():
            if score & value:
                self._allergens.append(key)

    def allergic_to(self, item):
        return item in self._allergens

    @property
    def lst(self):
        return self._allergens

File: solutions/cpp/allergies/1/allergies.cpp
#include "allergies.h"

#include <map>

namespace allergies {
    std::map<std::string, int> allergies = {
        {"eggs", 1},
        {"peanuts", 2},
        {"shellfish", 4},
        {"strawberries", 8},
        {"tomatoes", 16},
        {"chocolate", 32},
        {"pollen", 64},
        {"cats", 128}
    };

    allergy_test::allergy_test(const int score) : score_(score) {
    }

    bool allergy_test::is_allergic_to(const std::string& item) const {
        return (allergies[item] & score_) != 0;
    }

    std::unordered_set<std::string> allergy_test::get_allergies() const {
        std::unordered_set<std::string > result;
        for (const auto&[fst, snd] : allergies) {
            if ((snd & score_ )!= 0) {
                result.insert(fst);
            }
        }
        return result;
    }
} // namespace allergies
